packages/frontend/src/hooks/useQueryResults.ts

```diff
@@ -263,39 +268,63 @@ const getChartVersionResults = async (
  */
 const getQueryPaginatedResults = async (
     projectUuid: string,
-    data: PaginatedMetricQueryRequestParams,
+    data: ExecuteAsyncQueryRequestParams,
 ): Promise<ApiQueryResults & { queryUuid: string }> => {
-    const firstPage = await lightdashApi<ApiPaginatedQueryResults>({
+    const firstPage = await lightdashApi<ApiExecuteAsyncQueryResults>({
         url: `/projects/${projectUuid}/query`,
         version: 'v2',
         method: 'POST',
         body: JSON.stringify(data),
     });
 
     // Get all page rows in sequence
-    let allRows: ApiPaginatedQueryResults['rows'] = firstPage.rows;
-    let currentPage = firstPage;
-    while (currentPage.nextPage) {
-        currentPage = await lightdashApi<ApiPaginatedQueryResults>({
-            url: `/projects/${projectUuid}/query`,
+    let allRows: ResultRow[] = [];
+    let currentPage: ApiGetAsyncQueryResults | undefined;
+
+    while (
+        !currentPage ||
+        currentPage.status === WarehouseAsyncQueryStatus.PENDING ||
+        (currentPage.status === WarehouseAsyncQueryStatus.COMPLETED &&
+            currentPage.nextPage)
+    ) {
+        const page =
+            currentPage?.status === WarehouseAsyncQueryStatus.COMPLETED
+                ? currentPage?.nextPage
+                : 1;
+
+        currentPage = await lightdashApi<ApiGetAsyncQueryResults>({
+            url: `/projects/${projectUuid}/query/${firstPage.queryUuid}?page=${page}`,
             version: 'v2',
-            method: 'POST',
-            body: JSON.stringify({
-                queryUuid: firstPage.queryUuid,
-                page: currentPage.nextPage,
-            }),
+            method: 'GET',
+            body: undefined,
         });
-        allRows = allRows.concat(currentPage.rows);
+
+        const { status } = currentPage;
+
+        switch (status) {
+            case WarehouseAsyncQueryStatus.CANCELLED:
+                throw new Error('Query cancelled');
+            case WarehouseAsyncQueryStatus.ERROR:
+                throw new Error(currentPage.error || 'Query failed');
+            case WarehouseAsyncQueryStatus.COMPLETED:
+                allRows = allRows.concat(currentPage.rows);
+                break;
+            case WarehouseAsyncQueryStatus.PENDING:
```

Comment:
Add a timeout of 200 ms so it doesn't make a crazy amount of requests while pending.